### PR TITLE
core: several cleanups

### PR DESCRIPTION
**Pull request by YHNdnzj**
Some commits are split out from #32487 

Fixes https://github.com/systemd/systemd/issues/32512

### CODE REVIEW

File: src/core/namespace.c [hunk 1]
```diff
@@ -1029,49 +1030,47 @@ static int mount_private_dev(MountEntry *m, RuntimeScope scope) {
...
          * In nspawn and other containers it will be a symlink, in that case make it a symlink. */
         r = is_symlink("/dev/ptmx");
-        if (r < 0) {
-                log_debug_errno(r, "Failed to detect whether /dev/ptmx is a symlink or not: %m");
-                goto fail;
-        } else if (r > 0) {
+        if (r < 0)
+                return log_debug_errno(r, "Failed to detect whether /dev/ptmx is a symlink or not: %m");
+        if (r > 0) {
                 devptmx = strjoina(temporary_mount, "/dev/ptmx");
```
Comment (yuwata) [inline src/core/namespace.c]: Declaration of `devptmx` can be moved to in this block?

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by YHNdnzj** [comment]
Addresses the comment and modernized `mount_private_dev` a bit more, PTAL.

File: src/core/namespace.c [hunk 2]
```diff
@@ -976,7 +976,20 @@ static int clone_device_node(const char *node, const char *temporary_mount, bool
         return 0;
 }
 
-static char *settle_runtime_dir(RuntimeScope scope) {
+static int bind_mount_device_dir(const char *dir, const char *temporary_mount) {
```
Comment (yuwata) [inline src/core/namespace.c]: Not a matter, but I prefer to change the order of the argument, and rename `temporary_mount` -> `prefix` or so.

File: src/core/namespace.c [hunk 3]
```diff
@@ -976,7 +976,20 @@ static int clone_device_node(const char *node, const char *temporary_mount, bool
...
+        const char *t;
+
+        assert(dir);
+        assert(path_is_absolute(dir));
+        assert(temporary_mount);
+
+        t = strjoina(temporary_mount, dir);
```
Comment (yuwata) [inline src/core/namespace.c:986]: `prefix_roota()`? Then, we can drop `path_is_absolute(dir)`.

Comment (YHNdnzj) [reply src/core/namespace.c:986]: I'd like to ensure that the caller always passes absolute path. Also this is used below as bind mount src, so relative dir would be confusing?

Comment (yuwata) [reply src/core/namespace.c:986]: OK. Feel free to ignore.

**Review by yuwata — APPROVED** [review]

**Review by bluca — APPROVED** [review]

**Pull request merged**